crates/bevy_pbr/src/wireframe.rs

```diff
@@ -49,31 +67,169 @@ fn extract_wireframe_config(mut commands: Commands, wireframe_config: Res<Wirefr
     }
 }
 
-fn extract_wireframes(mut commands: Commands, query: Query<Entity, With<Wireframe>>) {
-    for entity in query.iter() {
-        commands.get_or_spawn(entity).insert(Wireframe);
+fn extract_wireframes(mut commands: Commands, query: Query<(Entity, Read<Wireframe>)>) {
+    for (entity, wireframe) in query.iter() {
+        commands.get_or_spawn(entity).insert(wireframe.clone());
     }
 }
 
-/// Controls whether an entity should rendered in wireframe-mode if the [`WireframePlugin`] is enabled
-#[derive(Component, Debug, Clone, Default, Reflect)]
+fn prepare_wireframes(
+    mut commands: Commands,
+    config: Res<WireframeConfig>,
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut wireframe_meta: ResMut<GlobalWireframeMeta>,
+    global_query: Query<(Entity, Option<&Wireframe>), (With<Handle<Mesh>>, With<MeshUniform>)>,
+    wireframe_query: Query<
+        (Entity, Option<&Wireframe>),
+        (With<Handle<Mesh>>, With<MeshUniform>, With<Wireframe>),
+    >,
+) {
+    wireframe_meta.uniforms.clear();
+    wireframe_meta.uniforms.push(WireframeUniform {
+        color: config.default_color.as_linear_rgba_f32().into(),
+    });
+
+    let add_wireframe_uniform = |(entity, wireframe): (Entity, Option<&Wireframe>)| {
+        let custom_color = wireframe.map(|wireframe| wireframe.color);
+        let uniform_offset = WireframeUniformOffset(if let Some(custom_color) = custom_color {
+            wireframe_meta.uniforms.push(WireframeUniform {
+                color: custom_color.as_linear_rgba_f32().into(),
+            })
+        } else {
+            0
+        });
+        commands.entity(entity).insert(uniform_offset);
+    };
+
+    if config.global {
+        global_query.for_each(add_wireframe_uniform);
+    } else {
+        wireframe_query.for_each(add_wireframe_uniform);
+    }
+
+    wireframe_meta
+        .uniforms
+        .write_buffer(&render_device, &render_queue);
+}
+
+pub struct GlobalWireframeBindGroup {
```

Comment:
Docs please; this is `pub`

Reply:
Done, altough these were "plugin internal" types. Maybe they should all be `#[doc(hidden)]`?

Reply:
Hmm, let's leave it like this for now. And thanks for the docs; they're still handy for developers.

Reply:
Actually, follow-up question: why are these `pub` at all?

Reply:
I made them to keep access to the internals open if needed, but they could be private I suppose.

Reply:
I think we should probably err towards making these private then. We can publicize them later as the API matures and/or users request it.